api/server/gin_middlewares.go

```diff
@@ -73,8 +73,8 @@ func loggerWrap(c *gin.Context) {
 	ctx, _ := common.LoggerWithFields(c.Request.Context(), extractFields(c))
 
 	if appName := c.Param(api.CApp); appName != "" {
-		c.Set(api.AppName, appName)
-		ctx = context.WithValue(ctx, api.AppName, appName)
+		c.Set(api.App, appName)
+		ctx = context.WithValue(ctx, api.App, appName)
```

Comment:
as noted earlier... this should be in addition to whatever we had really, and we really should use this moment to look up the app by name and shove the whole app in under `api.App` (and leave `AppName` for wider use)

Reply:
Actually, i disagree, because not all handlers should be covered with new middleware, that's why new code appears in new middleware that is applied to only small number of handlers.